services/motion/builtin/builtin.go

```diff
@@ -188,50 +190,18 @@ func (ms *builtIn) MoveOnMap(
 ) (bool, error) {
 	operation.CancelOtherWithLabel(ctx, builtinOpLabel)
 
-	// get the SLAM Service from the slamName
-	localizer, ok := ms.localizers[slamName]
-	if !ok {
-		return false, resource.DependencyNotFoundError(slamName)
-	}
-	ms.logger.Warn("This feature is currently experimental and does not support obstacle avoidance with SLAM maps yet")
-
-	// assert localizer as a slam service and get map limits
-	slamSvc, ok := localizer.(slam.Service)
-	if !ok {
-		return false, fmt.Errorf("cannot assert localizer of type %T as slam service", localizer)
-	}
-
-	// gets the extents of the SLAM map
-	limits, err := slam.GetLimits(ctx, slamSvc)
-	if err != nil {
-		return false, err
-	}
-
-	// create a KinematicBase from the componentName
-	component, ok := ms.components[componentName]
-	if !ok {
-		return false, resource.DependencyNotFoundError(componentName)
-	}
-	kw, ok := component.(base.KinematicWrappable)
-	if !ok {
-		return false, fmt.Errorf("cannot move component of type %T because it is not a KinematicWrappable Base", component)
-	}
-	kb, err := kw.WrapWithKinematics(ctx, localizer, limits)
+	// make call to motionplan
+	plan, kb, err := ms.PlanMoveOnMap(ctx, componentName, destination, slamName, extra)
 	if err != nil {
-		return false, err
+		return false, fmt.Errorf("error making plan for MoveOnMap: %v", err)
```

Comment:
the file has a lot of instances of using fmt.Errorf that someone else wrote, should I just change everything?

Reply:
Nah this isn't really important.  You can just keep this as is

Reply:
> I think errors.Errorf is the function we prefer last time I heard

Why the one over the other? I generally use `fmt.Errorf` and haven't heard otherwise

Reply:
To be honest I can't remember this was something that I was told like a year ago when I was trying to learn about the codebase.  I don't think it matters